tests/ut/test_decorators.py

```diff
@@ -197,6 +198,16 @@ class Dummy:
             assert memory_mock_cache.get.call_count == 0
             assert memory_mock_cache.set.call_count == 0
 
+    @pytest.mark.asyncio
+    async def test_cached_keeps_signature(self):
+        @cached()
+        async def what(self, a, b):
+            return "1"
+
+        assert what.__name__ == "what"
+        assert str(inspect.signature(what)) == '(self, a, b)'
+        assert inspect.getfullargspec(what.__wrapped__).args == ['self', 'a', 'b']
```

Comment:
@OOPman have a look at the behavior here. Dunno how zope does the interface check, but it should use the `signature` (https://docs.python.org/3/library/inspect.html#inspect.signature) one since it enables the `follow_wrapper` by default.

getfullargspec (https://docs.python.org/3/library/inspect.html#inspect.getfullargspec) doesn't support following the __wrapped__ attribute that `functools.wraps` adds.